lib/plug/crypto/message_verifier.ex

```diff
@@ -8,11 +8,13 @@ defmodule Plug.Crypto.MessageVerifier do
   tampered with.
   """
 
+  @delimiter "++"
+
   @doc """
   Decodes and verifies the encoded binary was not tampared with.
   """
   def verify(binary, secret) when is_binary(binary) and is_binary(secret) do
-    case String.split(binary, "--") do
+    case String.split(binary, @delimiter, parts: 2) do
```

Comment:
Let's not use parts because it would be broken anyway given we want the latest split on the delimiter and not the first one.